Utils/azure-explorer-common/test/java/com/microsoft/tooling/msservices/serviceexplorer/azure/rediscache/RedisCacheCreateIntegrationTest.java

```diff
@@ -0,0 +1,306 @@
+import com.microsoft.rest.RestClient;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import com.microsoft.tooling.IntegrationTestBase;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+import com.microsoft.tooling.msservices.helpers.UIHelper;
+import com.microsoft.tooling.msservices.serviceexplorer.Node;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener;
+import junit.framework.Assert;
+
+@PowerMockIgnore("javax.net.ssl.*")
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ AuthMethodManager.class, AzureManager.class, SubscriptionManager.class, DefaultLoader.class })
+
+public class RedisCacheCreateIntegrationTest extends IntegrationTestBase {
+
+    @Mock
+    private AuthMethodManager authMethodManagerMock;
+
+    @Mock
+    private AzureManager azureManagerMock;
+
+    @Mock
+    private SubscriptionManager subscriptionManagerMock;
+
+    @Mock
+    private UIHelper uiHelper;
+
+    private Azure azure;
+
+    private RedisCacheModule redisModule;
+
+    private String defaultSubscription;
+
+    private SubscriptionDetail currentsub;
+
+    private final LinkedHashMap<String, String> skus = RedisCacheUtil.initSkus();
+    private final LinkedHashMap<String, String> priceTier = initPriceTier();
+
+    RedisCreateConfig BasicNewResGrpConfig = new RedisCreateConfig("MonaC1BasicNew", "East US", "MonaC1BasicNewRg",
+            priceTier.get("BASIC1"), false, true);
+    RedisCreateConfig BasicNewResGrpConfigNonSsl = new RedisCreateConfig("MonaC2BasicNewNonSsl", "East US",
+            "MonaC2BasicNewNonSslRg", priceTier.get("BASIC2"), true, true);
+    RedisCreateConfig BasicExistedResGrpConfig = new RedisCreateConfig("MonaC3BasicExist", "East US", "MonaExist",
+            priceTier.get("BASIC3"), false, false);
+    RedisCreateConfig BasicExistedResGrpConfigNonSsl = new RedisCreateConfig("MonaC4BasicExistNonSsl", "East US",
+            "MonaExist", priceTier.get("BASIC4"), true, false);
+
+    RedisCreateConfig StdNewResGrpConfig = new RedisCreateConfig("MonaC3StdNew", "East US", "MonaC3StdNewRg",
+            priceTier.get("STD3"), false, true);
+    RedisCreateConfig StdNewResGrpConfigNonSsl = new RedisCreateConfig("MonaC4StdNewNonSsl", "East US",
+            "MonaC4StdNewNonSslRg", priceTier.get("STD4"), true, true);
+    RedisCreateConfig StdExistedResGrpConfig = new RedisCreateConfig("MonaC5StdExist", "East US", "MonaExist",
+            priceTier.get("STD5"), false, false);
+    RedisCreateConfig StdExistedResGrpConfigNonSsl = new RedisCreateConfig("MonaC6StdExistNonSsl", "East US",
+            "MonaExist", priceTier.get("STD1"), true, false);
+
+    RedisCreateConfig PremNewResGrpConfig = new RedisCreateConfig("MonaP1New", "East US", "MonaP1NewRg",
```

Comment:
for each creation strategy, we need to verify whether the final created rediscache really picked such configuration